### PR TITLE
[plugin_platform_interface] Update README and add a test

### PR DESCRIPTION
**Pull request by stuartmorgan**
Updates the example to use the new `verify` method, and makes discussion
of testing more generalized since it applies to uses like `Fake` as
well.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/plugin_platform_interface/README.md [hunk 1]
```diff
@@ -35,14 +35,15 @@ abstract class UrlLauncherPlatform extends PlatformInterface {
 This guarantees that UrlLauncherPlatform.instance cannot be set to an object that `implements`
 UrlLauncherPlatform (it can only be set to an object that `extends` UrlLauncherPlatform).
 
-## Mocking platform interfaces with Mockito
+## Mocking or faking platform interfaces
```
Comment (collinjackson) [inline packages/plugin_platform_interface/README.md:38]: You might want to mention this in the CHANGELOG as well.

**Review by collinjackson — APPROVED** [review]
One minor nit, LGTM otherwise

File: packages/plugin_platform_interface/README.md (hunk 1 above)
Comment (stuartmorgan) [reply packages/plugin_platform_interface/README.md:38]: Makes sense, done.

**Comment by collinjackson** [comment]
Here's an additional test for the old (wrong) behavior in case you want to include it in your update (or land separately): https://github.com/stuartmorgan/plugins/pull/1

I meant to push this to collinjackson/plugins and put up as a PR; have undone my change to your branch.

**Comment by stuartmorgan** [comment]
That seems like a good change to me; do you want to just push it back into this branch (with a CHANGELOG update) so we don't keep publishing new versions?

**Pull request merged**